packages/table/src/tableHotkeys.ts

```diff
@@ -295,19 +303,39 @@ export class TableHotkeys {
             return;
         }
 
+        const frozenRowsHeight = this.grid.getCumulativeHeightBefore(this.state.numFrozenRowsClamped);
+        const frozenColumnsWidth = this.grid.getCumulativeWidthBefore(this.state.numFrozenColumnsClamped);
+
+        // We align the scrollable container with non-frozen cells and the focused cell at the
+        // same origin: the end of the frozen columns and rows (`O`).
+        //            → scrollLeft
+        //       → cumulative cell widths
+        // * +---VIEWPORT RECT---+
+        // * |   | H | H | H | H |
+        // * +---+---+---+---+---+ ↓ cumulative row heights
+        // * | H | F | F | F | F |
+        // * +---+---O--SCROL. C.+
+        // * | H | F | S | S | S | ↓ scrollTop
+        // * +---+---+---+---+---+
+        // * | H | F | S | S | S |
+        // * +---+---+---+---+---+
+        // * | H | F | S | S | S |
+        // * +---+---+---+---+---+
+
         // sort keys in normal CSS position order (per the trusty TRBL/"trouble" acronym)
         // tslint:disable:object-literal-sort-keys
-        const viewportBounds = {
+        const scrollableContainerBounds = {
             top: viewportRect.top,
-            right: viewportRect.left + viewportRect.width,
-            bottom: viewportRect.top + viewportRect.height,
+            right: viewportRect.left + viewportRect.width - frozenColumnsWidth - this.state.rowHeaderWidth,
+            bottom: viewportRect.top + viewportRect.height - frozenRowsHeight - this.state.columnHeaderHeight,
             left: viewportRect.left,
         };
+
         const focusedCellBounds = {
-            top: this.grid.getCumulativeHeightBefore(row),
-            right: this.grid.getCumulativeWidthAt(col),
-            bottom: this.grid.getCumulativeHeightAt(row),
-            left: this.grid.getCumulativeWidthBefore(col),
+            top: this.grid.getCumulativeHeightBefore(row) - frozenRowsHeight,
+            right: this.grid.getCumulativeWidthAt(col) - frozenColumnsWidth,
+            bottom: this.grid.getCumulativeHeightAt(row) - frozenRowsHeight,
+            left: this.grid.getCumulativeWidthBefore(col) - frozenColumnsWidth,
```

Comment:
this change makes sense as going along with the change of `viewportBounds` -> `scrollableContainerBounds`